consensus/src/consensus/mod.rs

```diff
@@ -546,6 +569,9 @@ impl<N: Network> Future for Consensus<N> {
             }
         }
 
+        // Poll interval to wake up this task on a regular basis to perform head requests.
+        while self.head_requests_interval.poll_next_unpin(cx).is_ready() {}
```

Comment:
Is `while` desired here if only intended to register the waker for the interval task? Could simply be `let _ =`.